rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -1046,8 +1046,8 @@ class DebruijnInterpreter[M[_], F[_]](
             else ().pure[M]
         baseExpr <- evalSingleExpr(p)
         element  <- evalExpr(args.head)
-        result   <- delete(baseExpr, element)
         _        <- charge[M](REMOVE_COST) //TODO(mateusz.gorski): think whether deletion of an element from the collection should dependent on the collection type/size
+        result   <- delete(baseExpr, element)
```

Comment:
Wouldn't it make more sense to move the `charge` calls inside the functions they are charging for? Thanks to that such call won't be forgotten when one of those methods is called elsewhere

Reply:
That seems doable, will consider it the next time I break things in there :+1: